### PR TITLE
kube-state-metrics initial code

### PR DESCRIPTION
**Pull request by ghodss**

### CODE REVIEW

**Comment by ghodss** [comment]
@bgrant0607 this is ready for review

**Comment by bryk** [comment]
Does this have any potential for state metrics in a GUI? 

How can I access the metrics? 

I'm asking to understand the idea of this project.

**Comment by bgrant0607** [comment]
@ghodss @bryk 

The README could clarify this. This summarizes status information available via the API and exports it to monitoring systems, using Prometheus-compatible metrics.

The UI shouldn't need this. I could imagine, however, linking from dashboard to a monitoring dashboard. That could potentially be done via annotation conventions.

**Comment by bgrant0607** [comment]
FYI @a-robinson @vishh

**Comment by ghodss** [comment]
Updated the readme to include answers to @bryk's questions.

**Comment by vishh** [comment]
@ghodss Can you post a separate patch with just the scope and overall design?

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
-# kube-state-metrics
-Add-on agent to export cluster metrics to monitoring systems.
+# Overview
```
Comment (vishh) [inline README.md]: LGTM!

**Comment by vishh** [comment]
@ghodss LGTM

**Pull request merged**